UnityProject/Assets/Scripts/Core/Input System/InteractionV2/Validations.cs

```diff
@@ -684,6 +684,18 @@ public static bool HasUsedActiveWelder(Interaction interaction)
 		return welder.IsOn;
 	}
 
+	/// <summary>
+	/// Checks if an Emag has charges
+	/// </summary>
+	/// <param name="interaction"></param>
+	/// <returns></returns>
+	public static bool EmagHasCharges(GameObject toCheck)
```

Comment:
Hmm. I think a specific validation such as this should exist in the `Emag` component as most of the validations here are quite generic. I think the `HasUsedActiveWelder` validation is exceptional because it allows us to not check the component ourselves, but I think instance of `Emag` being used it seems like you need to get the component anyway, to consume a charge.